pkg/controller/machineset/controller_test.go

```diff
@@ -0,0 +1,194 @@
+package machineset
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"reflect"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"testing"
+	"time"
+)
+
+func TestMachineSetToMachines(t *testing.T) {
+	machineSetList := &v1alpha1.MachineSetList{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "MachineSetList",
+		},
+		Items: []v1alpha1.MachineSet{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "withMatchingLabels",
+					Namespace: "test",
+				},
+				Spec: v1alpha1.MachineSetSpec{
+					Selector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"foo": "bar",
+						},
+					},
+				},
+			},
+		},
+	}
+	controller := true
+	m := v1alpha1.Machine{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Machine",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "withOwnerRef",
+			Namespace: "test",
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					Name:       "Owner",
+					Kind:       "MachineSet",
+					Controller: &controller,
+				},
+			},
+		},
+	}
+	m2 := v1alpha1.Machine{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Machine",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "noOwnerRefNoLabels",
+			Namespace: "test",
+		},
+	}
+	m3 := v1alpha1.Machine{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Machine",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "withMatchingLabels",
+			Namespace: "test",
+			Labels: map[string]string{
+				"foo": "bar",
+			},
+		},
+	}
+	testsCases := []struct {
+		machine   v1alpha1.Machine
+		mapObject handler.MapObject
+		expected  []reconcile.Request
+	}{
+		{
+			machine: m,
+			mapObject: handler.MapObject{
+				Meta:   m.GetObjectMeta(),
+				Object: &m,
+			},
+			expected: []reconcile.Request{},
+		},
+		{
+			machine: m2,
+			mapObject: handler.MapObject{
+				Meta:   m2.GetObjectMeta(),
+				Object: &m2,
+			},
+			expected: nil,
+		},
+		{
+			machine: m3,
+			mapObject: handler.MapObject{
+				Meta:   m3.GetObjectMeta(),
+				Object: &m3,
+			},
+			expected: []reconcile.Request{
+				{NamespacedName: client.ObjectKey{Namespace: "test", Name: "withMatchingLabels"}},
+			},
+		},
+	}
+
+	v1alpha1.AddToScheme(scheme.Scheme)
+	r := &ReconcileMachineSet{
+		Client: newFakeClient(&m, &m2, &m3, machineSetList),
+		scheme: scheme.Scheme,
+	}
+
+	for _, tc := range testsCases {
+		got := r.MachineSetToMachines(tc.mapObject)
+		if !reflect.DeepEqual(got, tc.expected) {
+			t.Errorf("Case %s. Got: %v, expected: %v", tc.machine.Name, got, tc.expected)
+		}
+	}
+}
+
+func newFakeClient(initObjects ...runtime.Object) client.Client {
```

Comment:
Not sure this function buys you much, given it is just a one-line wrapper. And it only appears to be called from one place.